src/ui/server/helpers/percentage-number-validation/index.test.ts

```diff
@@ -0,0 +1,119 @@
+import percentageNumberValidation from '.';
+import { RequestBody, Errors } from '../../../types';
+import generateValidationErrors from '../validation';
+
+describe('server/helpers/percentage-number-validation', () => {
+  const mockErrors = {
+    summary: [],
+    errorList: {},
+  };
+
+  const mockBody = {
+    testField: '',
+  } as RequestBody;
+
+  const FIELD = 'testField';
+
+  const errorMessages = {
+    INCORRECT_FORMAT: 'Incorrect format',
+    BELOW_MINIMUM: 'Below minimum',
+    ABOVE_MAXIMUM: 'Above maximum',
+  } as Errors;
+
+  describe('percentage is a letter', () => {
+    it('should return a validation error', () => {
```

Comment:
Do we need test cases like percentage is `null`, `undefined` or `empty`?

Reply:
no, as that is done in the main validation file, this exclusively checks the inputted value if it exists

Reply:
Good point @Zainzzkk seems we should add `objectHasProperty` validation helper function to the actual validation? Then in the test we can just add "if field is empty" because the helper function handles the rest.

Reply:
it would mean that this could not be reused on any future optional percentage fields, is that fine?